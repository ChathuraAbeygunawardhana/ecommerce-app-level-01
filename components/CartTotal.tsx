import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

interface CartTotalProps {
  total: string;
}

const CartTotal = ({ total }: CartTotalProps) => {
  return (
    <View style={styles.totalContainer}>
      <View>
        <Text style={styles.totalText}>Total Amount:</Text>
      </View>
      <View>
        <Text style={styles.totalAmount}>${total}</Text>
      </View>
    </View>
  );
};

export default CartTotal;

const styles = StyleSheet.create({
  totalContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    padding: 16,
    borderTopColor: '#ddd',
    backgroundColor: '#fff',
  },
  totalText: {
    fontSize: 18,
  },
  totalAmount: {
    fontSize: 18,
    fontWeight: 'bold',
  },
});
