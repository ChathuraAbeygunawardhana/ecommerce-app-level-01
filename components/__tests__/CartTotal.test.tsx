import React from 'react';
import renderer from 'react-test-renderer';
import CartTotal from '../Cart/CartTotal';

describe('CartTotal', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CartTotal total="200" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});