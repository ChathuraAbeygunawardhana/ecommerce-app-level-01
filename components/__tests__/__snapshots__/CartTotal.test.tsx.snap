// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartTotal renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "borderTopColor": "#ddd",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 16,
    }
  }
>
  <View>
    <Text
      style={
        {
          "fontSize": 18,
        }
      }
    >
      Total Amount:
    </Text>
  </View>
  <View>
    <Text
      style={
        {
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      $
      200
    </Text>
  </View>
</View>
`;
